src/clients/scopes/AddScopeDialog.tsx

```diff
@@ -50,55 +100,207 @@ export const AddScopeDialog = ({
+                  const scopes = rows.map((scope) => ({ scope }));
+                  onAdd(scopes);
+                  toggleDialog();
+                }}
+              >
+                {t("common:add")}
+              </Button>,
+              <Button
+                id="modal-cancel"
+                key="cancel"
+                variant={ButtonVariant.link}
+                onClick={() => {
+                  setRows([]);
+                  toggleDialog();
+                }}
+              >
+                {t("common:cancel")}
+              </Button>,
+            ]
+          : [
+              <Dropdown
+                className="keycloak__client-scopes-add__add-dropdown"
+                id="add-dropdown"
+                key="add-dropdown"
+                direction={DropdownDirection.up}
+                isOpen={addToggle}
+                toggle={
+                  <DropdownToggle
+                    isDisabled={rows.length === 0}
+                    onToggle={() => setAddToggle(!addToggle)}
+                    isPrimary
+                    toggleIndicator={CaretUpIcon}
+                    id="add-scope-toggle"
+                  >
+                    {t("common:add")}
+                  </DropdownToggle>
+                }
+                dropdownItems={clientScopeTypesDropdown(t, action)}
+              />,
+              <Button
+                id="modal-cancel"
+                key="cancel"
+                variant={ButtonVariant.link}
+                onClick={() => {
+                  setRows([]);
+                  toggleDialog();
+                }}
+              >
+                {t("common:cancel")}
+              </Button>,
+            ]
+      }
     >
       <KeycloakDataTable
         loader={loader}
         ariaLabelKey="client-scopes:chooseAMapperType"
-        searchPlaceholderKey="client-scopes:searchFor"
+        searchPlaceholderKey={
+          filterType === "Name" ? "client-scopes:searchFor" : undefined
+        }
+        searchTypeComponent={
+          <Dropdown
+            onSelect={() => {
+              onFilterTypeDropdownSelect(filterType);
+            }}
+            data-testid="filter-type-dropdown"
+            toggle={
+              <DropdownToggle
+                id="toggle-id-9"
+                onToggle={onFilterTypeDropdownToggle}
+                toggleIndicator={CaretDownIcon}
+                icon={<FilterIcon />}
+              >
+                {filterType}
+              </DropdownToggle>
+            }
+            isOpen={isFilterTypeDropdownOpen}
+            dropdownItems={[
+              <DropdownItem
+                data-testid="filter-type-dropdown-item"
+                key="filter-type"
+              >
+                {filterType === "Name" ? t("protocol") : t("common:name")}
+              </DropdownItem>,
+            ]}
+          />
+        }
+        key={key}
+        toolbarItem={
+          filterType === "Protocol" && (
```

Comment:
You can use the enum to compare here as well.
```suggestion
          filterType === FilterType.Protocol && (
```